github/pulls_test.go

```diff
@@ -232,31 +234,63 @@ func TestPullRequestsService_Create_invalidOwner(t *testing.T) {
 }
 
 func TestPullRequestsService_Edit(t *testing.T) {
-	setup()
-	defer teardown()
+	tests := []struct {
+		input        *PullRequest
+		sendResponse string
+
+		wantUpdate   string
+		wantResponse *PullRequest
+	}{
+		{
+			input:        &PullRequest{Title: String("t")},
+			sendResponse: `{"number":1}`,
+			wantUpdate:   `{"title":"t"}`,
+			wantResponse: &PullRequest{Number: Int(1)},
+		},
+		{
+			// nil request
+			sendResponse: `{}`,
+			wantUpdate:   `{}`,
+			wantResponse: &PullRequest{},
+		},
+		{
+			// base update
+			input:        &PullRequest{Base: &PullRequestBranch{Ref: String("master")}},
+			sendResponse: `{"number":1,"base":{"ref":"master"}}`,
+			wantUpdate:   `{"base":"master"}`,
+			wantResponse: &PullRequest{
+				Number: Int(1),
+				Base:   &PullRequestBranch{Ref: String("master")},
+			},
+		},
+	}
 
-	input := &PullRequest{Title: String("t")}
+	for _, tt := range tests {
+		func() {
+			setup()
+			defer teardown()
```

Comment:
In fact, you can use the idea from `TestPullRequestsService_Merge_options` to remove the need for the nested closure with per-testcase `setup()/defer teardown()`. The idea there is to use `i` as the pull request ID, so that registered routes don't overlap.

See https://github.com/google/go-github/blob/c9c37fd6ce0247943d1aa34fc3813c62c5cb0225/github/pulls_test.go#L438-L486.